libraries/botbuilder/src/skills/skillHandler.ts

```diff
@@ -0,0 +1,185 @@
+/**
+ * @module botbuilder
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    Activity,
+    ActivityHandlerBase,
+    ActivityTypes,
+    BotAdapter,
+    ResourceResponse,
+    TurnContext
+} from 'botbuilder-core';
+import { AuthenticationConfiguration, ICredentialProvider, ClaimsIdentity } from 'botframework-connector';
+
+import { ChannelServiceHandler } from '../channelServiceHandler';
+import { SkillConversationIdFactoryBase } from './skillConversationIdFactoryBase';
+import { BotFrameworkAdapter } from '../botFrameworkAdapter';
+
+/**
+ * A Bot Framework Handler for skills.
+ */
+export class SkillHandler extends ChannelServiceHandler {
+    public readonly SkillConversationReferenceKey: string = 'SkillConversationReference';
+    /**
+     * Initializes a new instance of the SkillHandler class.
+     * @param adapter An instance of the BotAdapter that will handle the request.
+     * @param bot The ActivityHandlerBase instance.
+     * @param conversationIdFactory A SkillConversationIdFactoryBase to unpack the conversation ID and map it to the calling bot.
+     * @param credentialProvider The credential provider.
+     * @param authConfig The authentication configuration.
+     * @param channelService The string indicating if the bot is working in Public Azure or in Azure Government (https://aka.ms/AzureGovDocs).
+     */
+    constructor(
+        private readonly adapter: BotAdapter,
+        private readonly bot: ActivityHandlerBase,
+        private readonly conversationIdFactory: SkillConversationIdFactoryBase,
```

Comment:
I think this parameter is required, should we add a null check?

Reply:
There's a falsy check on line 49